tests/LinkDotNet.Blog.UnitTests/Domain/BlogPostTests.cs

```diff
@@ -24,6 +24,21 @@ public void ShouldUpdateBlogPost()
         blogPostToUpdate.PreviewImageUrlFallback.Should().Be("Url2");
         blogPostToUpdate.IsPublished.Should().BeTrue();
         blogPostToUpdate.Tags.Should().BeNullOrEmpty();
+        blogPostToUpdate.Slug.Should().NotBeNull();
+    }
+
+    [Theory]
+    [InlineData("blog title","blog-title")]
+    [InlineData("blog      title", "blog-title")]
+    [InlineData("blog +title", "blog-title")]
+    [InlineData("blog/title", "blogtitle")]
+    [InlineData("blog /title", "blog-title")]
+    public void ShouldGenerateValidSlug(string title, string expectedSlug)
+    {
+        var blogPost = BlogPost.Create(title, "Desc", "Content", "Url", true, previewImageUrlFallback: "Url2");
```

Comment:
For ease of use, you can use `BlogPostBuilder` - that will always create a valid BlogPost and is more concise.
```csharp
var blogPost = new BlogPostBuilder().WithTitle(title).Build()M
```